lib/galaxy/datatypes/images.py

```diff
@@ -528,6 +528,47 @@ def sniff_prefix(self, file_prefix):
         return False
 
 
+class Star(data.Text):
+    """Base format class for Relion STAR (Self-defining
+    Text Archiving and Retrieval) image files.
+    https://relion.readthedocs.io/en/latest/Reference/Conventions.html"""
+    file_ext = "star"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        """Set the peek and blurb text"""
+        if not dataset.dataset.purged:
+            dataset.peek = data.get_file_peek(dataset.file_name)
+            dataset.blurb = 'Relion STAR data'
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def sniff(self, filename):
+        """Each file must have one or more data blocks.
+        The start of a data block is defined by the keyword
+        "data_" followed by an optional string for
+        identification (e.g., "data_images").  All text
+        before the first "data_" keyword are comments
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('1.star')
+        >>> Star().sniff(fname)
+        True
+        >>> fname = get_test_fname('interval.interval')
+        >>> Star().sniff(fname)
+        False
+        """
+        with open(filename) as fh:
```

Comment:
Can you find a more modern example data type using ``@build_sniff_from_prefix`` and ``sniff_prefix`` so we can avoid re-opening this file.

I'm also worry this logic is going to pick up a lot of files - I'd prefer something tighter. Is it possible to look for a ``loop_`` on its own line also?

Reply:
Thanks @jmchilton, I think (er, hope) I've got these items covered in my 2nd commit.  The STAR format is tricky because only a single data_ block seems to be required - loop_ blocks and such are optional.  The enhanced checks are much better than the original though.  :)

Please let me know if you feel more is necessary.  The file name itself must have a .star extension if that can be used in the sniffer.